libpod/boltdb_state.go

```diff
@@ -1352,6 +1400,16 @@ func (s *BoltState) AddVolume(volume *Volume) error {
 		return errors.Wrapf(err, "error marshalling volume %s config to JSON", volume.Name())
 	}
 
+	// Volume state is allowed to not exist
+	var volStateJSON []byte
+	if volume.state != nil {
+		stateJSON, err := json.Marshal(volume.state)
+		if err != nil {
+			return errors.Wrapf(err, "error marshalling volume %s state to JSON", volume.Name())
+		}
+		volStateJSON = stateJSON
```

Comment:
any reason not to volStateJSON, err = json.Marshal?

Reply:
I think `json.Marshal` on a nil pointer makes angry errors

Reply:
Oh wait, i see what you're saying. Probably will work.